cannon/mipsevm/versions/version_test.go

```diff
@@ -15,3 +15,39 @@ func TestParseStateVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSupported(t *testing.T) {
+	type TestCase struct {
+		name     string
+		input    int
+		expected bool
+	}
+	cases := []TestCase{}
+
+	maxSupportedValue := 0
+	for _, ver := range StateVersionTypes {
+		versionValue := int(ver)
+		if versionValue > maxSupportedValue {
+			maxSupportedValue = versionValue
+		}
+		if IsSupportedMultiThreaded64(ver) {
+			cases = append(cases, TestCase{name: "Supported version " + ver.String(), input: versionValue, expected: true})
+		} else {
+			cases = append(cases, TestCase{name: "Unsupported version " + ver.String(), input: versionValue, expected: false})
+		}
```

Comment:
Might it be better to craft these by hand?
The implementation of `IsSupported` is almost that of `IsSupportedMultiThreaded64`, so it's barely testing much in practise.

Reply:
I generally agree it's better not to have too much logic in tests.  But because we update these vm versions a lot, it's kind of annoying to remember to update a bunch of constants in tests.  On balance, I think the lower maintenance burden is worth the trade-off here. This approach also ensures we're checking every defined version.